worker/src/main/java/org/apache/celeborn/service/deploy/worker/storage/PartitionFilesSorter.java

```diff
@@ -656,14 +646,41 @@ private void deleteOriginFiles() throws IOException {
       }
     }
 
-    private ByteBuffer expandBufferAndUpdateMemoryTracker(int oldCapacity, int newCapacity)
-        throws InterruptedException {
-      memoryManager.releaseSortMemory(oldCapacity);
-      memoryManager.reserveSortMemory(newCapacity);
```

Comment:
if newCapacity > maxSortMemoryRatio, will cause  `sortMemoryReady()` return false and this memory can't release, then cause this thread stuck here and cause PartitionSorter can't take FileSorter to executor.